### PR TITLE
Cretzman multiple bugs 01 17 2023

### PR DESCRIPTION
**Pull request by michaelcretzman**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ x] Issues
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

- CDS-50596	Reference expressions for output from Jenkins vs Shell Script
- Bug	CDS-48449	Update Docs for ServiceV2 and Env Runtime Input Support
- Bug	CDS-48459	Update Terraform version on docs
- Bug	CDS-47102	Update the docs with the Cloudformation Tags syntax - user made an error when configuring cloudformation with tags

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Review by thisrohangupta — APPROVED** [review]
The Changes for Cloudformation looks good.

**Review by Debanitrkl — APPROVED** [review]

File: docs/continuous-delivery/cd-execution/cd-general-steps/run-jenkins-jobs-in-cd-pipelines.md [hunk 1]
```diff
@@ -92,15 +88,35 @@ In **Advanced**, you can use the following options:
 
 For Harness to capture Jenkins environment variables, your Jenkins configuration requires the [EnvInject Plugin](https://wiki.jenkins.io/display/JENKINS/EnvInject+Plugin). The plugin does not provide full compatibility with the Pipeline Plugin. See [Known Incompatibilities](https://wiki.jenkins.io/display/JENKINS/EnvInject+Plugin#EnvInjectPlugin-Knownincompatibilities) from Jenkins.Harness captures certain environment variables from the Jenkins build.
```
Comment (douglas-j-bothwell) [inline docs/continuous-delivery/cd-execution/cd-general-steps/run-jenkins-jobs-in-cd-pipelines.md:89]:
```suggestion
For Harness to capture Jenkins environment variables, your Jenkins configuration requires the [EnvInject Plugin](https://wiki.jenkins.io/display/JENKINS/EnvInject+Plugin). The plugin does not provide full compatibility with the Pipeline plugin. See [Known Incompatibilities](https://wiki.jenkins.io/display/JENKINS/EnvInject+Plugin#EnvInjectPlugin-Knownincompatibilities) from Jenkins. Harness captures certain environment variables from the Jenkins build.
```

File: docs/continuous-delivery/cd-execution/cd-general-steps/run-jenkins-jobs-in-cd-pipelines.md [hunk 2]
```diff
@@ -92,15 +88,35 @@ In **Advanced**, you can use the following options:
...
+For example, `<+execution.steps.[step group Id].steps.[step Id].build.jobUrl>`.
+
+:::
 
-* **Job Status:** `<+pipeline.stages.<stage Id>.spec.execution.steps.<step Id>.build.executionStatus>`
-* **Job URL:** `<+pipeline.stages.<stage Id>.spec.execution.steps.<step Id>.build.jobUrl>`
-* **Build number:** `<+pipeline.stages.<stage Id>.spec.execution.steps.<step Id>.build.buildNumber>`
-* **Build display name:** `<+pipeline.stages.<stage Id>.spec.execution.steps.<step Id>.build.buildDisplayName>`
-* **Full build display name:** `<+pipeline.stages.<stage Id>.spec.execution.steps.<step Id>.build.buildFullDisplayName>`
+You can reference this Job information is subsequent steps in your pipeline or another Pipeline.
```
Comment (douglas-j-bothwell) [inline docs/continuous-delivery/cd-execution/cd-general-steps/run-jenkins-jobs-in-cd-pipelines.md:117]:
```suggestion
You can reference this Job information in subsequent steps in your pipeline or in another pipeline.
```

File: docs/continuous-delivery/cd-execution/cd-general-steps/run-jenkins-jobs-in-cd-pipelines.md [hunk 3]
```diff
@@ -92,15 +88,35 @@ In **Advanced**, you can use the following options:
...
 
-You can reference this Job information is subsequent steps in your Pipeline or another Pipeline.
+To reference it in another pipeline, you would use the full pipeline path to the build information. For example, `<+pipeline.stages.[stage_Id].spec.execution.steps.[step_Id].build.executionStatus>`.
```
Comment (douglas-j-bothwell) [inline docs/continuous-delivery/cd-execution/cd-general-steps/run-jenkins-jobs-in-cd-pipelines.md:119]:
```suggestion
To reference the Job information in another pipeline, use the full pipeline path to the build information. For example, `<+pipeline.stages.[stage_Id].spec.execution.steps.[step_Id].build.executionStatus>`.
```

File: docs/continuous-delivery/cd-execution/cd-general-steps/using-shell-scripts.md [hunk 4]
```diff
@@ -71,9 +71,9 @@ In **Script Input Variables**, you simply select **Expression** and paste the ex
 
 ![](./static/using-shell-scripts-18.png)
 
-In the Script, you declare the variable using the **Name** value.
+In the Script, you declare the variable using the **Name** value (in this example, `foo`).
 
-![](./static/using-shell-scripts-19.png)
+![picture 3](static/3efd3f47e73c3ca4804bd0e728d8815194ae80c9284ddfe0c11fb07c520b3b0c.png)
 
 At deployment runtime, Harness will evaluate the expression and the variable will contain its output.
```
Comment (douglas-j-bothwell) [inline docs/continuous-delivery/cd-execution/cd-general-steps/using-shell-scripts.md:78]:
```suggestion
At deployment runtime, Harness evaluates the expression and the variable contains its output.
```

**Review by douglas-j-bothwell — DISMISSED** [review]
A few typos but otherwise looks good!

File: docs/continuous-delivery/cd-execution/cd-general-steps/run-jenkins-jobs-in-cd-pipelines.md [hunk 5]
```diff
@@ -92,15 +88,35 @@ In **Advanced**, you can use the following options:
...
 
-* **Job Status:** `<+pipeline.stages.<stage Id>.spec.execution.steps.<step Id>.build.executionStatus>`
-* **Job URL:** `<+pipeline.stages.<stage Id>.spec.execution.steps.<step Id>.build.jobUrl>`
-* **Build number:** `<+pipeline.stages.<stage Id>.spec.execution.steps.<step Id>.build.buildNumber>`
-* **Build display name:** `<+pipeline.stages.<stage Id>.spec.execution.steps.<step Id>.build.buildDisplayName>`
-* **Full build display name:** `<+pipeline.stages.<stage Id>.spec.execution.steps.<step Id>.build.buildFullDisplayName>`
+You can reference this Job information in subsequent steps in your pipeline or in another pipeline.
 
-You can reference this Job information is subsequent steps in your Pipeline or another Pipeline.
+To reference the Job information in another pipeline, use the full pipeline path to the build information. For example, `<+pipeline.stages.[stage_Id].spec.execution.steps.[step_Id].build.executionStatus>`.
```
Comment (brettzu) [inline docs/continuous-delivery/cd-execution/cd-general-steps/run-jenkins-jobs-in-cd-pipelines.md:119]: This is to reference in another stage of the same pipeline, not in another pipeline.

File: docs/continuous-delivery/cd-execution/cd-general-steps/using-shell-scripts.md [hunk 6]
```diff
@@ -38,7 +38,7 @@ In the stage, in **Execution**, click **Add Step**.
 
 Select **Shell Script**.
 
-Enter a name for the step. An Id is generated. This Id identifies the step and is used in variable expressions. For example, if the Id is **Shell Script**, the expression might be `<+steps.Shell_Script.output.outputVariables.myvar>`.
+Enter a name for the step. An Id is generated. This Id identifies the step and is used in variable expressions. For example, if the Id is **Shell Script**, the expression might be `<+execution.steps.Shell_Script.output.outputVariables.myvar>`.
```
Comment (brettzu) [inline docs/continuous-delivery/cd-execution/cd-general-steps/using-shell-scripts.md:41]: It works either way. The "execution." part seems to be optional.

File: docs/continuous-delivery/cd-execution/cd-general-steps/using-shell-scripts.md [hunk 7]
```diff
@@ -97,31 +97,28 @@ In **Name**, enter a name to use in other steps that will reference this variabl
 The format to reference the output variable can be one of the following:
 
 * Within the stage:
-	+ Referencing the step output:
-		- `<+steps.[step_id].output.outputVariables.[output_variable_name]>`.
-	+ Referencing the step output execution:
-		- `<+execution.steps.[step_id].output.outputVariables.[output_variable_name]>`
-* Anywhere in the Pipeline:
-	+ `<+pipeline.stages.[stage_Id].spec.execution.steps.[step_id].output.outputVariables.[output_variable_name]>`
+	+ `<+execution.steps.[step_id].output.outputVariables.[output_variable_name]>`
```
Comment (brettzu) [inline docs/continuous-delivery/cd-execution/cd-general-steps/using-shell-scripts.md:100]: The "execution." is optional, so either way works. Do we want to mention both or only one form?

File: docs/continuous-delivery/cd-execution/cd-general-steps/using-shell-scripts.md [hunk 8]
```diff
@@ -138,7 +135,7 @@ Next, you reference that output variable as a secret, like this:
 
 
 ```
-echo "my secret: " <+steps.CreateScript.output.outputVariables.myvar>
+echo "my secret: " <+execution.steps.CreateScript.output.outputVariables.myvar>
```
Comment (brettzu) [inline docs/continuous-delivery/cd-execution/cd-general-steps/using-shell-scripts.md:138]: execution optional

**Review by brettzu — CHANGES_REQUESTED** [review]

File: docs/continuous-delivery/cd-execution/cd-general-steps/using-shell-scripts.md (hunk 7 above)
Comment (michaelcretzman) [reply docs/continuous-delivery/cd-execution/cd-general-steps/using-shell-scripts.md:100]: we should mention only execution. it helps them understand where the variable is evaluated.

**Comment by michaelcretzman** [comment]
Prasad said I can merge this.

**Pull request merged**